UnityProject/Assets/Scripts/Tilemaps/Behaviours/Layers/Matrix.cs

```diff
@@ -311,6 +311,27 @@ public IEnumerable<RegisterTile> GetRegisterTile(Vector3Int localPosition, bool
 		return (isServer ? ServerObjects : ClientObjects).Get(localPosition);
 	}
 
+	public IEnumerable<T> GetAs<T>(Vector3Int localPosition, bool isServer) where T : class
```

Comment:
Could you also comment this to say GetAs is only to be used if T is a class which inherit from RegisterTile

Reply:
```suggestion
	public IEnumerable<T> GetAs<T>(Vector3Int localPosition, bool isServer) where T : RegisterTile
```

Actually would this work instead, but add a comment anyway?

Reply:
done